src/main/java/uk/gov/companieshouse/account/validator/service/maintenance/AccountMaintenanceService.java

```diff
@@ -34,17 +43,36 @@ public static boolean isEmptyOrNull(Collection<?> collection) {
     }
 
     public void deleteCompleteSubmissions() {
-        logger.debug("Inside deleteFiles method");
+        LocalDateTime now = LocalDateTime.now();
+        LocalDate date = getBoundaryDate(now, DAYS_TO_DELETE);
+        Map<String, Object> infoContext = new HashMap<>();
+        infoContext.put("Deletion to (exclusive)", date);
+        infoContext.put("Deletion requested at", now);
+        logger.info("Deletion date range for old accounts", infoContext);
         try {
-            List<RequestStatus> completeRequestStatusList = statusRepository.findByStatus(RequestStatus.STATE_COMPLETE);
+            List<RequestStatus> completeRequestStatusList = statusRepository
+                    .findByStatusAndModifiedDateTimeLessThan(RequestStatus.STATE_COMPLETE, date);
             if (!isEmptyOrNull(completeRequestStatusList)) {
                 completeRequestStatusList.forEach(requestStatus -> deleteRequest(requestStatus.fileId()));
             }
+            List<RequestStatus> nullCreatedDateRequestStatuses = statusRepository
+                    .findByStatusAndCreatedDateTimeIsNull(RequestStatus.STATE_COMPLETE);
+            if (!isEmptyOrNull(nullCreatedDateRequestStatuses)) {
+                nullCreatedDateRequestStatuses.forEach(requestStatus -> deleteRequest(requestStatus.fileId()));
+            }
```

Comment:
I think it would be cleaner to combine these lists before performing the delete, that way the null check is completed once.